### ISSUE TITLE
feat: Brick generating bricks

### ISSUE THREAD

**Issue by Fasuh** [felangel/mason#265]
**Description**

I'd like to have a brick, which generates bricks. Those inner bricks have their own variables, separate from the main brick, and also have some variables which the main bricks injects inside them.

Right now when i try to make a brick like that, everything goes smoothly until it reaches a filename with lambdas and a variable.
(screenshot)

I think it would be nice to have a way of escaping variables and lambdas so they are ignored by the generator and placed without the escape character.

**Requirements**

- Making it possible to escape variables and lambdas so they dont generate as empty Strings

**Additional Context**

![image](https://user-images.githubusercontent.com/39801418/155035637-e891b2df-8d5a-488c-afcf-a03f3c889f4c.png)

**Comment by felangel** [comment]
Hi @Fasuh 👋 
Thanks for opening an issue!

You should be able to use `{{{my_variable}}}` to indicate it should be unescaped. If that doesn't help it would be great if you could provide a link to a minimal reproduction sample, thanks! 🙏

**Comment by Fasuh** [comment]
Hi, should i use it as 
`{{{ {{#snakeCase}}{{name}}{{/snakeCase}} }}}` ?

**Comment by felangel** [comment]
> Hi, should i use it as
> `{{{ {{#snakeCase}}{{name}}{{/snakeCase}} }}}` ?

No you should be good to just use:
`{{#snakeCase}}{{{name}}}{{/snakeCase}} `

**Comment by Fasuh** [comment]
> > Hi, should i use it as
> > `{{{ {{#snakeCase}}{{name}}{{/snakeCase}} }}}` ?
> 
> No you should be good to just use: `{{#snakeCase}}{{{name}}}{{/snakeCase}} `

if i enter that in a file and generate it it just returns an empty file

**Comment by felangel** [comment]
> > > Hi, should i use it as
> > > `{{{ {{#snakeCase}}{{name}}{{/snakeCase}} }}}` ?
> > 
> > 
> > No you should be good to just use: `{{#snakeCase}}{{{name}}}{{/snakeCase}} `
> 
> if i enter that in a file and generate it it just returns an empty file

Can you provide a link to a minimal reproduction sample that illustrates the issue with some reproduction steps?

**Comment by Fasuh** [comment]
i just composed a repo for that
https://github.com/Fasuh/test_repo_mason
if you run `mason make test` it will generate an empty file `test` despite it having an escaped variable

**Comment by felangel** [comment]
> i just composed a repo for that
> https://github.com/Fasuh/test_repo_mason
> if you run `mason make test` it will generate an empty file `test` despite it having an escaped variable

Thanks I’ll have a look shortly 👍

**Comment by felangel** [comment]
@Fasuh can you provide a bit more context regarding what you’re trying to achieve?

**Comment by Fasuh** [comment]
> @Fasuh can you provide a bit more context regarding what you’re trying to achieve?

Of course, i'd like to have a project template brick which i'd hold on my git repository, this brick would contain all of our dependencies and stuff we require to start the project, this brick takes arguments i.e. project name.

This brick would also be responsible for making local bricks folder. This folder contains our local bricks which should always be saved per project because they could change in time, and we want them to be still compatible with the project they were initialized in, so it has to be local.

basically i want this project template to be able to generate bricks folder which contains some smaller basic bricks with their own variables which are completely separate from the main template variables.
